---
title: "TDEseq"
author: "Fan Yue"
date: "2023/7/24"
output: html_document
 # prettydoc::html_pretty:
 #   theme: cayman
 #   highlight: github
---

## Overview
TDEseq is implemented as an open source R package for detecting genes with temporal dynamic expression patterns in time-series scRNA-seq transcriptomic studies. TDEseq models the relationship between log normalized data and the corresponding time points through constrained additive mixed model and can detect the DE genes as well as its temporal dynamic pattern simultaneously.

## Installation
```
library(devtools)
install_github("fanyue322/TDEseq")
```

## Usage
TDEseq starts with the normalized gene expression scRNA-seq data and the corresponding time points.

### Application on mouse liver development data
```
library(Seurat)
library(TDEseq)
```
### Loading data
We used a sample mouse liver development during embyro process scRNA-seq data as a main example which contains expression measurement of 14,226 genes on 345 cells. This dataset can be downloaded from their original study GSE90047. We also saved the processed data that we used in our examples in RData format, which can be downloaded from [here]( https://drive.google.com/file/d/1354UjYPgx492l8rxjPvW8UB0LRPjD0tR/view?usp=sharing).
```
load('GSE90047.rda')
time=seurat@meta.data$stage
expData=seurat@assays$RNA@data
```
### Detecting temporal DE genes through TDEseq
TDEseq takes the normalized data and time points as inputs, and output the p-value and shape information for each gene.
```
res=TDEseq(expData,time,LMM=FALSE)
```

By default, TDEseq uses the Linear version. Alternatively, one can perform the mixed version of TDEseq, which accounts for the cell-cell relatedness information within an identical individual, which can be set by `LMM` argument, e.g. : 
```
group=seurat@meta.data$orig.ident
res=TDEseq(expData,time,group=group,LMM=TRUE)
```

### Including covariates
TDEseq also supports the inclusion of covariates in DE analysis. For example, age, sex, etc.
```
res=TDEseq(expData,time,group=group,zmat=pseudotime,LMM=TRUE)
```

### Parameter settings
In `TDEseq()`, besides the default parameters, users can also set other parameters as below to filter genes or cells:

`logFC_threshold`: Limit testing to genes which show the maximum on average X-fold difference (log-scale) between any two time points. Default is 0.0.

`pct`: Only test genes that are detected in a minimum fraction of pct cells. Default is 0.1.

`threshold`: Only return markers that have an FDR(BY adjusted) < return.thresh. Default is 0.05.

`max_cells_per_ident`: Down sample each identity class to a max number. Default is no downsampling. 

`min_cells_per_timepoints`:Minimum number of cells in one of the time points. 

`LMM`:Denotes which methods to use. Default is FALSE(Linear TDEseq). Set `LMM=TRUE` to perform Mixed TDEseq.



