---
title: "TDEseq"
output: html_document
 # prettydoc::html_pretty:
 #   theme: cayman
 #   highlight: github
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Overview
TDEseq is implemented as an open source R package for detecting genes with temporal dynamic expression patterns in time-series scRNA-seq transcriptomic studies. TDEseq models the relationship between log normalized data and the corresponding time points through constrained additive mixed model and can detect the DE genes as well as its temporal dynamic pattern simultaneously.

## Installation
```{r eval=FALSE, include=TRUE}
library(devtools)
install_github("fanyue322/TDEseq")
```

## Usage

### Application on mouse liver development data
```{lib eval=FALSE, include=TRUE}
library(Seurat)
library(TDEseq)
```
### Loading data
We used a sample 151673 in DLPFC data as a main example which contains expression measurement of 14,226 genes on 345 cells. This dataset can be downloaded from their original study GSE90047. We also saved the processed data that we used in our examples in RData format, which can be downloaded from [here]( https://drive.google.com/drive/folders/1YKYLGeoZYZO0EBIzzoa3PBPSx-4UgXkr).
```{lib eval=FALSE, include=TRUE}
load('GSE90047.rda')
time=seurat@meta.data$stage
expData=seurat@assays$RNA@data
```
### Detecting temporal DE genes through TDEseq
TDEseq takes the normalized data and time points as inputs, and output the p-value and shape information for each gene.
```{lib eval=FALSE, include=TRUE}
res=TDEseq(expData,time,LMM=FALSE)
```

Alternatively, one can perform the mixed version of TDEseq, which accounts for the cell-cell relatedness information within an identical individual.
```{lib eval=FALSE, include=TRUE}
group=seurat@meta.data$orig.ident
res=TDEseq(expData,time,group=group,LMM=TRUE)
```

### Including covariates
TDEseq also supports the inclusion of covariates in DE analysis. For example, the pseudotime for each cell.
```{lib eval=FALSE, include=TRUE}
res=TDEseq(expData,time,group=group,z=pseudotime,LMM=TRUE)
```
