<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>TDEseq</title>

<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>





<link rel="stylesheet" href="install_files/style.css" type="text/css" />





</head>

<body>




<section class="page-header">
<h1 class="title toc-ignore project-name">TDEseq</h1>
</section>



<section class="main-content">
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>TDEseq is implemented as an open source R package for detecting genes with temporal dynamic expression patterns in time-series scRNA-seq transcriptomic studies. TDEseq models the relationship between log normalized data and the corresponding time points through constrained additive mixed model and can detect the DE genes as well as its temporal dynamic pattern simultaneously.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<pre><code>library(devtools)
install_github(&quot;fanyue322/TDEseq&quot;)</code></pre>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<div id="application-on-mouse-liver-development-data" class="section level3">
<h3>Application on mouse liver development data</h3>
<pre><code>library(Seurat)
library(TDEseq)</code></pre>
</div>
<div id="loading-data" class="section level3">
<h3>Loading data</h3>
<p>We used a sample 151673 in DLPFC data as a main example which contains expression measurement of 14,226 genes on 345 cells. This dataset can be downloaded from their original study GSE90047. We also saved the processed data that we used in our examples in RData format, which can be downloaded from <a href="https://drive.google.com/drive/folders/1YKYLGeoZYZO0EBIzzoa3PBPSx-4UgXkr">here</a>.</p>
<pre><code>load(&#39;GSE90047.rda&#39;)
time=seurat@meta.data$stage
expData=seurat@assays$RNA@data</code></pre>
</div>
<div id="detecting-temporal-de-genes-through-tdeseq" class="section level3">
<h3>Detecting temporal DE genes through TDEseq</h3>
<p>TDEseq takes the normalized data and time points as inputs, and output the p-value and shape information for each gene.</p>
<pre><code>res=TDEseq(expData,time,LMM=FALSE)</code></pre>
<p>Alternatively, one can perform the mixed version of TDEseq, which accounts for the cell-cell relatedness information within an identical individual.</p>
<pre><code>group=seurat@meta.data$orig.ident
res=TDEseq(expData,time,group=group,LMM=TRUE)</code></pre>
</div>
<div id="including-covariates" class="section level3">
<h3>Including covariates</h3>
<p>TDEseq also supports the inclusion of covariates in DE analysis. For example, the pseudotime for each cell.</p>
<pre><code>res=TDEseq(expData,time,group=group,z=pseudotime,LMM=TRUE)</code></pre>
</div>
</div>
</section>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
